### PR TITLE
.NET Dependency Updates

### PR DESCRIPTION
**Pull request by pmachapman**
This change updates .NET dependencies, and fixes .NET compiler warnings and removes all pragma warning suppressions. Changes include:
* Updated ParatextData to 9.3 and SIL.Core to 10.1
* Modified `MockScrText` to conform to `ScrText` more correctly
* Updated Hangfire and Hangfire.Mongo to the latest version, and modified the config to match the behavior changes in the new versions
* Updated MongoDB Driver to 2.18 (adds support for MongoDB 6.0), and updated dependent code
* Updated the ASP.Net Core dependencies to the latest 6.0.x version
* Changed the NetCore.MailKit dependency to MailKit, as we were using none of the extra functionality provided by NetCore.MailKit
* Updated Jering.Javascript.NodeJS, Autofac, EdjCase.JsonRpc.Router, HtmlAgilityPack, NSubstitute, NUnit, IdentityModel, idunno.Authentication.Basic to the latest stable versions
* Modified the Solution file to reference Visual Studio 2022 (cosmetic change to Solution icon)

This change is [Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1636)

### CODE REVIEW

**Review by Nateowami — COMMENTED** [review]
Quick question about this. I briefly checked out this branch and then went back to master. I notice that afterwards I get this error:
```
Hangfire.Mongo version: 0.6.6.0, uses a schema prior to the current database.
Backwards migration is not supported. Please resolve this manually (e.g. by dropping the database).
Please see https://github.com/sergeyzwezdin/Hangfire.Mongo#migration for further information.
```

(I didn't even run `dotnet restore`)

I'm guessing this is just a Hangfire schema version and has nothing to do with the version of Mongo. But it did remind me that we're running an old version of Mongo (4.0.28). We need to update but just haven't done so yet. Can you confirm that all of these dependency updates should continue to work with the version of Mongo we're currently running? If not some of them may need to wait until after we update Mongo.

**Review by pmachapman — COMMENTED** [review]
> I'm guessing this is just a Hangfire schema version and has nothing to do with the version of Mongo.

Yes, that is correct. If we downgrade from this release, we will need to delete the Hangfire schema.

> Can you confirm that all of these dependency updates should continue to work with the version of Mongo we're currently running?

Yes, they will run correctly. MongoDB C#/.NET Driver 2.18 requires MongoDB 3.6 or higher. See https://mongodb.github.io/mongo-csharp-driver/2.18/what_is_new/#what-s-new-in-2-14-0 for details. I downgraded my version from 6.0.3 to 4.0.28 to confirm, and all worked correctly.

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1636#-:-NJ1eCvX1-mPBMIg3tSR:b-i9y1hv)* status: 0 of 16 files reviewed, all discussions resolved

**Review by Nateowami — COMMENTED** [review]
> Reviewed 1 of 12 files at r3.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1636)* status: 1 of 16 files reviewed, 2 unresolved discussions (waiting on @github-code-scanning[bot] and @pmachapman)

___
*[`xForge.sln` line 3 at r4](https://reviewable.io/reviews/sillsdev/web-xforge/1636#-NJ863F6CjTvaxkyNA-m:-NJ863F6CjTvaxkyNA-n:brc4g3t) ([raw file](https://github.com/sillsdev/web-xforge/blob/fdab892955b7018e25da65f9f3950d2d9a408628/xForge.sln#L3)):*
> ```
> 
> Microsoft Visual Studio Solution File, Format Version 12.00
> # Visual Studio Version 17
> ```

@nigel-wells Are you using Visual Studio code? If so, is this going to impact you?

**Review by nigel-wells — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1636)* status: 1 of 16 files reviewed, 2 unresolved discussions (waiting on @github-code-scanning[bot], @Nateowami, @nigel-wells, and @pmachapman)

___
*[`xForge.sln` line 3 at r4](https://reviewable.io/reviews/sillsdev/web-xforge/1636#-NJ863F6CjTvaxkyNA-m:-NJ8C4kL4g6VCiT8aUsP:b-dblppq) ([raw file](https://github.com/sillsdev/web-xforge/blob/fdab892955b7018e25da65f9f3950d2d9a408628/xForge.sln#L3)):*
<details><summary><i>Previously, Nateowami (Nathaniel Paulus) wrote…</i></summary><blockquote>

@nigel-wells Are you using Visual Studio code? If so, is this going to impact you?
</blockquote></details>

I'm running VS 2022 - 17.4.2 so should be fine. I only use it for reviewing various solutions rather than building anything though.

File: src/SIL.XForge/Realtime/RealtimeJsonService.cs [hunk 1]
```diff
@@ -14,30 +14,26 @@ namespace SIL.XForge.Realtime
     /// </summary>
     public class RealtimeJsonService : IJsonService
     {
-        private static readonly JsonSerializer _serializer = new JsonSerializer
+        private static readonly JsonSerializer Serializer = new JsonSerializer
         {
             ContractResolver = new CamelCasePropertyNamesContractResolver(),
-            NullValueHandling = NullValueHandling.Ignore
+            NullValueHandling = NullValueHandling.Ignore,
```
Comment (Nateowami) [inline src/SIL.XForge/Realtime/RealtimeJsonService.cs:20]: Is the trailing comma a deliberate style change?

**Review by Nateowami — COMMENTED** [review]
Just a couple questions.

**Review by Nateowami — COMMENTED** [review]
I forgot to mention I'm trying out reviewing from within VS Code (which allows quickly hovering to look up method documentation, among other things).

It worked well for the most part, except the diff for `MockScrText.cs` failed to show up.

**Review by pmachapman — COMMENTED** [review]
Yes, I tested the UpdateHelp program locally, and it ran correctly. The update to HtmlAgilityPack is a very minor bug fix release.

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1636#-:-NJC6nBR-JttLIg089P5:bwwp4gf)* status: 1 of 16 files reviewed, 2 unresolved discussions (waiting on @github-code-scanning[bot] and @Nateowami)

___
*[`src/SIL.XForge/Realtime/RealtimeJsonService.cs` line 20 at r6](https://reviewable.io/reviews/sillsdev/web-xforge/1636#-NJ1dp2qOXZnKmoGCvAq-r6-20:-NJC7Hx03vu08hyQFljW:botozkf) ([raw file](https://github.com/sillsdev/web-xforge/blob/891a0003c71be3b8ebe68e9e9bbafc8075fe32ed/src/SIL.XForge/Realtime/RealtimeJsonService.cs#L20)):*
<details><summary><i>Previously, Nateowami (Nathaniel Paulus) wrote…</i></summary><blockquote>

Is the trailing comma a deliberate style change?
</blockquote></details>

Yes. It is a style we can configure in `.editorconfig`. I have personally found it a very useful style change, as it means that adding a new argument or property only changes on line in the git diff, as opposed to two.
___
*[`test/SIL.XForge.Scripture.Tests/Services/MockScrText.cs` line 25 at r4](https://reviewable.io/reviews/sillsdev/web-xforge/1636#-NJ1dp2qOXZnKmoGCvAs-r4-25:-NJ8En7v6OfOrUsZaZOb:b-896fix) ([raw file](https://github.com/sillsdev/web-xforge/blob/fdab892955b7018e25da65f9f3950d2d9a408628/test/SIL.XForge.Scripture.Tests/Services/MockScrText.cs#L25)):*
<details><summary><i>Previously, github-code-scanning[bot] wrote…</i></summary><blockquote>

## Virtual call in constructor or destructor

Avoid virtual calls in a constructor or destructor.

[Show more details](https://github.com/sillsdev/web-xforge/security/code-scanning/291)
</blockquote></details>

Done.

**Review by Nateowami — APPROVED** [review]
:lgtm:

> Reviewed 1 of 12 files at r3, 2 of 6 files at r6.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1636#-:-NJCDRseBT72GAJjm5Z1:bnfp4nl)* status: 4 of 16 files reviewed, 1 unresolved discussion (waiting on @github-code-scanning[bot])

**Pull request merged**